src/org/traccar/notification/EventForwarder.java

```diff
@@ -43,26 +47,39 @@ public EventForwarder() {
     private static final String KEY_GEOFENCE = "geofence";
     private static final String KEY_DEVICE = "device";
 
-    public void forwardEvent(Event event, Position position) {
+    public final void forwardEvent(Event event, Position position) {
 
         BoundRequestBuilder requestBuilder = Context.getAsyncHttpClient().preparePost(url);
+        requestBuilder.setBodyEncoding(StandardCharsets.UTF_8.name());
+
+        requestBuilder.addHeader("Content-Type", getContentType());
 
-        requestBuilder.addHeader("Content-Type", "application/json; charset=utf-8");
         if (!header.equals("")) {
-            String[] headerLines = header.split("\\r?\\n");
-            for (String headerLine: headerLines) {
-                String[] splitedLine = headerLine.split(":", 2);
-                if (splitedLine.length == 2) {
-                    requestBuilder.setHeader(splitedLine[0].trim(), splitedLine[1].trim());
-                }
-            }
+            FluentCaseInsensitiveStringsMap params = new FluentCaseInsensitiveStringsMap();
+            params.putAll(splitParams(header, ":"));
+            requestBuilder.setHeaders(params);
         }
 
-        requestBuilder.setBody(preparePayload(event, position));
+        setContent(event, position, requestBuilder);
         requestBuilder.execute();
     }
 
-    private byte[] preparePayload(Event event, Position position) {
+    protected Map<String, List<String>> splitParams(String params, String separator) {
```

Comment:
Why is this in the base class?

Reply:
method name was not so accurate, it splits http headers and request params, renamed